buildSrc/src/main/java/org/elasticsearch/gradle/Distribution.java

```diff
@@ -20,23 +20,66 @@
 
 public enum Distribution {
 
-    INTEG_TEST("integ-test", "zip"),
-    ZIP("elasticsearch", "zip"),
-    ZIP_OSS("elasticsearch-oss", "zip");
+    INTEG_TEST("integ-test"),
+    DEFAULT("elasticsearch"),
+    OSS("elasticsearch-oss");
 
     private final String fileName;
-    private final String fileExtension;
 
-    Distribution(String name, String fileExtension) {
+    Distribution(String name) {
         this.fileName = name;
-        this.fileExtension = fileExtension;
     }
 
-    public String getFileName() {
+    public String getArtifactName() {
         return fileName;
     }
 
     public String getFileExtension() {
-        return fileExtension;
+        if (this.equals(INTEG_TEST)) {
+            return "zip";
+        } else {
+            switch (getOS()) {
+                case LINUX:
+                case MAC:
+                    return "tar.gz";
+                case WINDOWS:
+                    return "zip";
+                default:
+                    throw new IllegalStateException("Can't determine extensions for " + getOS());
+            }
+        }
+    }
+
+    public String getClassifier() {
+        switch (getOS()) {
+            case LINUX:
+                return "linux-x86_64";
+            case MAC:
+                return "darwin-x86_64";
+            case WINDOWS:
+                return "windows-x86_64";
+            default:
+                throw new IllegalStateException("Can't determine extensions for " + getOS());
+        }
+    }
+
+    private enum OS {
+        WINDOWS,
+        MAC,
+        LINUX
+    }
+
+    private OS getOS() {
```

Comment:
This is common functionality that tends to get reimplemented a lot. We rely on the Gradle internal `OperatingSystem` class a bunch of other places for this. We should either 1) just continue to use that type since it's so pervasive or 2) provide a replacement and use it consistently across `buildSrc`. We can do this as a separate PR.

Reply:
I agree, we could move this in it's own class as needed, I just wanted to avoid relying on Gradle classes here.